### ISSUE TITLE
Handle missing numeric values for ScriptSort

### ISSUE THREAD

**Issue by cbuescher** [elastic/elasticsearch#17190]
There is an open Todo in ScriptSortParser concerned with handling missung values for the Numeric script type case. Currently it seems we are sorting missing values last when order is ascending, but first when the order is descending:

https://github.com/elastic/elasticsearch/blob/5107388fe9/core/src/main/java/org/elasticsearch/search/sort/ScriptSortParser.java#L185

After a brief talk with @jpountz it seems like we should probably also sort them last when order is descending.

**Comment by cbuescher** [comment]
@clintongormley might also have an opinion about this?

**Comment by jpountz** [comment]
+1 on sorting missing values last by default for all fields

**Comment by clintongormley** [comment]
For normal sort, we support the `missing` parameter to determine whether missing values should be sorted `_first`, `_last`, or a custom value. Can we not do the same here?

**Comment by jpountz** [comment]
I had not even noticed that script sorting does not support this parameter. I guess the assumption is that if special handling of missing values is needed then the script can do it by itself.

**Comment by cbuescher** [comment]
I think we cannot support the `missing` parameter here, as @jpountz mentioned the script needs to handle those cases itself. I just played around with tests and read the code, and it appears that numeric scripts (at least for groovy, didn't check the others) always return 0.0d (because of `double runAsDouble()`), even if the script acesses a null field in the doc. I also thinks this makes the original question from the TODO I linked to irrelevant and I'd close this issue if nobody objects.
